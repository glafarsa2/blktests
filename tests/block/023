#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2018 Omar Sandoval
#
# Smoke test all null_blk queue modes. Regression test for patch "null_blk: fix
# zoned support for non-rq based operation".

. tests/block/rc
. common/null_blk

DESCRIPTION="do I/O on all null_blk queue modes"
QUICK=1

requires() {
	_have_null_blk
}

test() {
	echo "Running ${TEST_NAME}"

	local queue_mode
	for ((queue_mode = 0; queue_mode <= 2; queue_mode++)); do
		if _init_null_blk gb=1 queue_mode="$queue_mode"; then
			echo "Queue mode $queue_mode"
			dd if=/dev/nullb0 of=/dev/null iflag=direct bs=64k status=none
			dd if=/dev/null of=/dev/nullb0 oflag=direct bs=64k status=none
			_exit_null_blk
		fi
	done

	echo "Test complete"
}
