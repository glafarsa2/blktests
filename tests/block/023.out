Running block/023
Queue mode 0
Queue mode 2
Test complete
